### PR TITLE
Update Documentation

### PR DESCRIPTION
**Pull request by joepetrowski**
Closes https://github.com/paritytech/substrate/issues/2159

Updates:

**Balances**

- Switch to relative links
- Update examples to reflect current usage after https://github.com/paritytech/substrate/pull/2048

**Contract**

- Style, formatting, misspellings

**Staking**

- Added links so that it renders on rust stable
- Enforced max line length of 120 characters

**Sudo & System**

- Made common sections consistent with other modules

**Timestamp**

- Made common sections consistent with other modules
- Updated relative links

### CODE REVIEW

File: srml/balances/src/lib.rs [hunk 1]
```diff
@@ -16,29 +16,29 @@
 
 //! # Balances Module
 //!
-//! The balances module provides functionality for handling accounts and balances. To use the balances module, you need
-//! to implement the [balances Trait](https://crates.parity.io/srml_balances/trait.Trait.html). Supported dispatchables
-//! are documented in the [`Call` enum](https://crates.parity.io/srml_balances/enum.Call.html).
+//! The balances module provides functionality for handling accounts and balances.
+//! To use it in your module, you need to implement the balances [`Trait`](./trait.Trait.html).
```
Comment (gavofyork) [inline srml/balances/src/lib.rs]:
```suggestion
//! To use it in your runtime, you need to implement the balances [`Trait`](./trait.Trait.html).
```

File: srml/balances/src/lib.rs [hunk 2]
```diff
@@ -16,29 +16,29 @@
...
-//! - Lookup of an index to reclaim an account
-//! - Managing total issuance
-//! - Setting and managing locks
+//! - Getting and setting free balances.
+//! - Retrieving total, reserved and unreserved balances.
+//! - Repatriating a reserved balance to a beneficiary account that exists.
+//! - Transferring a balance between accounts (when not reserved).
+//! - Slashing an account balance.
+//! - Account creation and removal.
+//! - Looking up an index to reclaim an account.
```
Comment (gavofyork) [inline srml/balances/src/lib.rs]: Remove.

File: srml/balances/src/lib.rs [hunk 3]
```diff
@@ -60,90 +61,71 @@
...
 //! 	let gas_left = gas_meter.gas_left();
 //!
 //! 	// Increase total spent gas.
 //! 	<GasSpent<T>>::mutate(|block_gas_spent| *block_gas_spent += gas_spent);
 //!
-//! 	let refund = <T::Gas as As<T::Balance>>::as_(gas_left) * gas_meter.gas_price;
-//! 	// Refund gas using balances module
-//! 	let refund_imbalance = <balances::Module<T>>::deposit_creating(transactor, refund);
-//! 	// Handle imbalance
+//! 	// Refund gas left by the price it was bought.
```
Comment (gavofyork) [inline srml/balances/src/lib.rs]:
```suggestion
//! 	// Refund gas left by the price it was bought at.
```

File: srml/contract/src/lib.rs [hunk 4]
```diff
@@ -17,11 +17,11 @@
 //! # Contract Module
 //!
 //! The contract module provides functionality for the runtime to deploy and execute WebAssembly smart-contracts.
-//! The supported dispatchable functions are documented as part of the [`Call`](./enum.Call.html) enum.
+//! To use it in your module, you need to implement the contracts [`Trait`](./trait.Trait.html).
```
Comment (gavofyork) [inline srml/contract/src/lib.rs]:
```suggestion
//! To use it in your runtime, you need to implement the contracts [`Trait`](./trait.Trait.html).
```

File: srml/contract/src/lib.rs [hunk 5]
```diff
@@ -17,11 +17,11 @@
...
 //!
 //! ## Overview
 //!
-//! This module extends accounts (see `Balances` module) to have smart-contract functionality.
+//! This module extends accounts (see the balances module) to have smart-contract functionality.
```
Comment (gavofyork) [inline srml/contract/src/lib.rs]: Not true. It just relies on currency so it can be used with any module that implements that.

File: srml/contract/src/lib.rs [hunk 6]
```diff
@@ -33,7 +33,7 @@
 //! This call can alter the storage entries of the smart-contract account, create new smart-contracts,
 //! or call other smart-contracts.
 //!
-//! Finally, when the `Balances` module determines an account is dead (i.e. account balance fell below the
+//! Finally, when the balances module determines an account is dead (i.e. account balance fell below the
```
Comment (gavofyork) [inline srml/contract/src/lib.rs]: Not true - it's now decoupled from the balances module.

File: srml/balances/src/lib.rs (hunk 1 above)
Comment (joepetrowski) [reply srml/balances/src/lib.rs]: Will update in each file.

File: srml/contract/src/lib.rs [hunk 7]
```diff
@@ -122,38 +122,38 @@ pub trait ComputeDispatchFee<Call, Balance> {
...
+	/// The size of stored value in octet.
 	pub storage_size: u64,
 }
 
-/// Get a trie id (trie id must be unique and collision resistant depending upon its context)
-/// Note that it is different than encode because trie id should have collision resistance
-/// property (being a proper uniqueid).
+/// Get a trie id (trie id must be unique and collision resistant depending upon its context).
+/// Note that it is different than encode because trie id should be collision resistant
+/// (being a proper uniqueid).
```
Comment (gavofyork) [inline srml/contract/src/lib.rs]:
```suggestion
/// (being a proper unique identifier).
```

File: srml/contract/src/lib.rs [hunk 8]
```diff
@@ -122,38 +122,38 @@ pub trait ComputeDispatchFee<Call, Balance> {
...
 pub trait TrieIdGenerator<AccountId> {
```
Comment (gavofyork) [inline srml/contract/src/lib.rs]: A bit out of scope, but I'm not sure this should be named "Trie". The trie is an abstraction level beyond storage and is an implementation detail as far as the runtime module is concerned. In this case, it's specifically the `AccountId` that must be a unique identifier, so it should probably be called `AccountIdGenerator`

cc @pepyakin

File: srml/system/src/lib.rs [hunk 9]
```diff
@@ -168,15 +168,15 @@ pub trait Trait: 'static + Eq + Clone {
 	/// The aggregated event type of the runtime.
 	type Event: Parameter + Member + From<Event>;
 
-	/// A piece of information which can be part of the digest (as a digest item).
+	/// A piece of information thqt can be part of the digest (as a digest item).
```
Comment (gavofyork) [inline srml/system/src/lib.rs]:
```suggestion
	/// A piece of information that can be part of the digest (as a digest item).
```

File: srml/system/src/lib.rs [hunk 10]
```diff
@@ -404,8 +404,8 @@ impl<T: Trait> Module<T> {
 		let storage_root = T::Hashing::storage_root();
 		let storage_changes_root = T::Hashing::storage_changes_root(parent_hash, number.as_() - 1);
 
-		// we can't compute changes trie root earlier && put it to the Digest
-		// because it will include all currently existing temporaries
+		// We can't compute changes trie root earlier && put it to the Digest
```
Comment (gavofyork) [inline srml/system/src/lib.rs]: inconsistent with above change. to be honest, there's really no need to mess around with internal comments for algorithms. if the coder is happy with all-lowercase, then that's fine.

File: srml/staking/src/lib.rs [hunk 11]
```diff
@@ -17,51 +17,57 @@
...
-//! are chosen based upon those who voluntarily place funds under deposit. Under deposit, those funds are rewarded under
-//! normal operation but are held at pain of _slash_ (expropriation) should the staked maintainer be found not to be
-//! discharging their duties properly.
+//! The staking module is the means by which a set of network maintainers (known as _authorities_ in some contexts
+//! and _validators_ in others) are chosen based upon those who voluntarily place funds under deposit. Under deposit,
+//! those funds are rewarded under normal operation but are held at pain of _slash_ (expropriation) should the
+//! staked maintainer be found not to be discharging its duties properly.
 //!
-//! You can start using the Staking module by implementing the staking [`Trait`].
+//! You can start using the staking module by implementing the staking [`Trait`](./trait.Trait.html).
```
Comment (gavofyork) [inline srml/staking/src/lib.rs]: It's a name so it deserves capitalisation.
```suggestion
//! You can start using the Staking module by implementing the `staking::` [`Trait`](./trait.Trait.html).
```

File: srml/staking/src/lib.rs [hunk 12]
```diff
@@ -106,43 +113,44 @@
 //!
 //! ## Interface
 //!
-//! ### Dispatchable
+//! ### Dispatchable Functions
 //!
-//! The Dispatchable functions of the staking module enable the steps needed for entities to accept and change their
-//! role, alongside some helper functions to get/set the metadata of the module.
+//! The dispatchable functions of the staking module enable the steps needed for entities to accept and change their
+//! role, along with some helper functions to get/set the metadata of the module.
```
Comment (gavofyork) [inline srml/staking/src/lib.rs]:
```suggestion
//! role, alongside some helper functions to get/set the metadata of the module.
```

**Comment by joepetrowski** [comment]
Fixed review comments with the exception of [`Trie/AccountId` naming](https://github.com/paritytech/substrate/pull/2172#discussion_r271588577).

Have changed "To use it in your module..." to "To use it in your runtime..." for modules except the System module, as I think it applies more to modules than runtimes. Correct?

File: srml/contract/src/lib.rs [hunk 13]
```diff
@@ -16,14 +16,14 @@
 
 //! # Contract Module
 //!
-//! The contract module provides functionality for the runtime to deploy and execute WebAssembly smart-contracts.
-//! The supported dispatchable functions are documented as part of the [`Call`](./enum.Call.html) enum.
```
Comment (bkchr) [inline srml/contract/src/lib.rs]: Why did you remove the `Call` enum here?

File: srml/contract/src/lib.rs [hunk 14]
```diff
@@ -16,14 +16,14 @@
...
+//! To use it in your runtime, you need to implement the [`contracts::Trait`](./trait.Trait.html).
 //!
 //! ## Overview
 //!
-//! This module extends accounts (see `Balances` module) to have smart-contract functionality.
-//! These "smart-contract accounts" have the ability to create smart-contracts and make calls to other contract
-//! and non-contract accounts.
+//! This module extends accounts based on the `Currency` trait to have smart-contract functionality. It can
+//! be used with other modules that implement accounts based on `Currency`. These "smart-contract accounts"
+//! have the ability to create smart-contracts and make calls to other contract and non-contract accounts.
```
Comment (bkchr) [inline srml/contract/src/lib.rs]:
```suggestion
//! have the ability to create smart-contracts and make calls to other contracts and non-contract accounts.
```

File: srml/sudo/src/lib.rs [hunk 15]
```diff
@@ -18,31 +18,29 @@
...
-//! The sudo module allows for a single account (called the "sudo key")
+//! The Sudo module allows for a single account (called the "sudo key")
 //! to execute dispatchable functions that require a `Root` call
 //! or designate a new account to replace them as the sudo key.
 //! Only one account can be the sudo key at a time.
 //!
-//! You can start using the sudo module by implementing the sudo [`Trait`].
-//!
-//! Supported dispatchable functions are documented in the [`Call`] enum.
+//! You can start using the Sudo module by implementing the [`sudo::Trait`](./trait.Trait.html).
```
Comment (bkchr) [inline srml/sudo/src/lib.rs]: Here you also dropped `Call`?

File: srml/sudo/src/lib.rs [hunk 16]
```diff
@@ -103,7 +101,8 @@
 //!
 //! ## Genesis Config
 //!
-//! To use the sudo module, you need to set an initial superuser account as the sudo `key`.
+//! The Balances module depends on the [genesis configuration](./struct.GenesisConfig.html).
```
Comment (bkchr) [inline srml/sudo/src/lib.rs]:
```suggestion
//! The Balances module depends on the [`GenesisConfig`](./struct.GenesisConfig.html).
```

File: srml/contract/src/lib.rs (hunk 13 above)
Comment (joepetrowski) [reply srml/contract/src/lib.rs]: I moved it to the Dispatchable Functions section so that you see a link from the brief summaries to the more detailed documentation.
